### PR TITLE
Refactored `Direction8` class to not use interfaces and removed the o…

### PR DESCRIPTION
**Pull request by EvidentlyCube**
…ther obsolete direction classes

### CODE REVIEW

File: src.common/Enums/Direction8.ts [hunk 1]
```diff
@@ -1,14 +1,35 @@
-export interface Direction8 {
-	readonly id: number;
-	readonly name: string;
-	readonly x: number;
-	readonly y: number;
-	readonly nextCW: Direction8;
-	readonly nextCCW: Direction8;
-	readonly opposite: Direction8;
-}
+export class Direction8 implements Direction8 {
```
Comment (kieranmillar) [inline src.common/Enums/Direction8.ts]: This class now implements ... itself? Is this still necessary?

File: src.common/Enums/Direction8.ts [hunk 2]
```diff
@@ -1,14 +1,35 @@
...
+	public static readonly UpLeft: Readonly<Direction8> = new Direction8(1, "upleft");
```
Comment (kieranmillar) [inline src.common/Enums/Direction8.ts]: For my benefit, what's the implications of using `static readonly` vs `const`?

File: src.common/Enums/Direction8.ts (hunk 1 above)
Comment (EvidentlyCube) [reply src.common/Enums/Direction8.ts]: Whoopsie, missed it

File: src.common/Enums/Direction8.ts (hunk 2 above)
Comment (EvidentlyCube) [reply src.common/Enums/Direction8.ts]: in TypeScript you can't have class-level constants. If you replaced `static readonly` with `const` it wouldn't compile. But it essentially does what `const` would do - create a staticly accessible variable that can't be changed.

**Pull request merged**